source/transfer_list.rst

```diff
@@ -391,23 +408,60 @@ Inputs:
    `te_base_addr` with `tl.checksum`.
 
 #. If an existing XFERLIST_VOID TE was chosen to be overwritten in step 1, and
-   `old_void_data_size - new_data_size` is greater or equal to `0x8`:
+   `old_void_data_size - new_data_size` is greater or equal to `0x8` then call
+   `Adding a void TE`_ with following arguments:
+
+   #. `void.te.base_addr` = `te_base_addr + align8(new_data_size + 0x8)`
+
+   #. `void.te.size` =  `old_void_data_size - align8(new_data_size)`
+
+Removing a TE
+^^^^^^^^^^^^^
+
+Inputs:
+
+- `tl_base_addr`: Base address of the TL from which TE to be deleted
+- `tag_id`: ID number of the tag to be deleted
+- `scrub_data`: Boolean option to scrub TE data
+
+#. Validate the TL header as mentioned in `Validating a TL header`_
+
+#. Locate the `te_base_addr` of the TE to be removed by iterating through the TL.
+
+#. Mark the TE as void by setting its tag_id to XFERLIST_VOID
+
+#. Erase the TE data if `scrub_data` is TRUE
+
+#. If checksum is enabled, recalculate and update `tl.checksum`
+
+Overwriting a TE
+^^^^^^^^^^^^^^^^
+
+Inputs:
+
+- `tl_base_addr`: Base address of the TL from which TE to be overwritten
+- `tag_id`: ID number of the tag to be overwritten
+- `new_data_size`: Size in bytes of the data to be encapsulated in the TE
+- [data]: Data to be copied into the TE
+
+#. Validate the TL header as mentioned in :ref:`Validating a TL header`
+
+#. Locate the `te_base_addr` of the TE to be overwritten by iterating through the TL
+
+#. Ensure that `te.data_size` is greater or equal to `new_data_size`
 
-   #. Use `te_base_addr + align8(new_data_size + 0x8)` as the new `te_base_addr`
-      for a new XFERLIST_VOID tag.
+#. Erase the old TE data from `te_base_addr + 0x8` until  `te.data_size`
 
-   #. If `has_checksum`, xor the 8 bytes from `te_base_addr` to
-      `te_base_addr + 0x8` with `tl.checksum`.
+#. Copy the TE data into `te_base_addr + 0x8`
 
-   #. Set `te.tag_id` (`te_base_addr + 0x0`) to `0x0` (XFERLIST_VOID).
+#. If `te.data_size - new_data_size` is greater or equal to `0x8` then call
+   `Adding a void TE`_ with following arguments:
 
-   #. Set `te.hdr_size` (`te_base_addr + 0x3`) to `0x8`.
+   #. `void.te.base_addr` = `te_base_addr + align8(new_data_size + 0x8)`
 
-   #. Set `te.data_size` (`te_base_addr + 0x4`) to
-      `old_void_data_size - align8(new_data_size) - 0x8`.
+   #. `void.te.size` =  `te.data_size - align8(new_data_size)`
 
-   #. If `has_checksum`, xor the 8 bytes from `te_base_addr` to
-      `te_base_addr + 0x8` with `tl.checksum`.
+#. If checksum is enabled, recalculate and update `tl.checksum`
```

Comment:
This should again explicitly describe how to account for the new bytes after making the change.